homeassistant/components/zwave_js/update.py

```diff
@@ -0,0 +1,190 @@
+        """Initialize a Z-Wave device firmware update entity."""
+        self.driver = driver
+        self.node = node
+        self.available_firmware_updates: list[FirmwareUpdateInfo] = []
+        self._latest_version_firmware: FirmwareUpdateInfo | None = None
+
+        name: str = (
+            node.name or node.device_config.description or f"Node {node.node_id}"
+        )
+        # Entity class attributes
+        self._attr_name = f"{name}: Firmware"
+        self._base_unique_id = get_valueless_base_unique_id(driver, node)
+        self._attr_unique_id = f"{self._base_unique_id}.firmware_update"
+        # device is precreated in main handler
+        self._attr_device_info = DeviceInfo(
+            identifiers={get_device_id(driver, node)},
+        )
+
+        self._attr_installed_version = node.firmware_version
+
+    @property
+    def extra_state_attributes(self) -> Mapping[str, Any] | None:
+        """Return entity specific state attributes."""
+
+        return {
+            "available_firmware_updates": [
+                {
+                    "version": firmware.version,
+                    "release_notes": firmware.changelog,
+                }
+                for firmware in self.available_firmware_updates
+            ]
+        }
+
+    async def async_update(self) -> None:
+        """Update the entity."""
+        self.available_firmware_updates = (
+            await self.driver.controller.async_get_available_firmware_updates(
+                self.node, API_KEY_FIRMWARE_UPDATE_SERVICE
+            )
+        )
+        self._async_process_updates(False)
+
+    @callback
+    def _async_process_updates(self, write_state: bool = True) -> None:
+        """Process updates."""
+        if self.available_firmware_updates:
+            self._latest_version_firmware = firmware = get_latest_firmware(
+                self.available_firmware_updates
+            )
+            self._attr_latest_version = firmware.version
+            self._attr_release_summary = firmware.changelog
+        else:
+            self._latest_version_firmware = None
+            self._attr_latest_version = None
+            self._attr_release_summary = None
+        if write_state:
+            self._async_write_ha_state()
+
+    async def async_install(
+        self, version: str | None, backup: bool, **kwargs: Any
+    ) -> None:
+        """Install an update."""
+        if not self._latest_version_firmware:
+            raise HomeAssistantError("No firmware update available")
+
+        firmware = None
+        if version is None:
+            firmware = self._latest_version_firmware
+        else:
+            for firmware_ in self.available_firmware_updates:
+                if firmware_.version == version:
+                    firmware = firmware_
+                    break
+        if firmware is None:
+            raise HomeAssistantError(f"Version {version} not found")
```

Comment:
If it's a user input error we normally raise ValueError.
```suggestion
            raise ValueError(f"Version {version} not found")
```

Reply:
should we `raise HomeAssistantError from ValueError(...)`? If I recall that's how the service UI picks up the error

Reply:
We don't want users to be able to ignore bad input.

Reply:
Understood!